drivers/bluetooth/hci/hci_stm32wba.c

```diff
@@ -380,7 +394,91 @@ static int bt_hci_stm32wba_open(const struct device *dev, bt_hci_recv_t recv)
 	return ret;
 }
 
+#if defined(CONFIG_BT_HCI_SETUP)
+
+bt_addr_t *bt_get_ble_addr(void)
+{
+	bt_addr_t *bd_addr;
+	uint32_t udn;
+	uint32_t company_id;
+	uint32_t device_id;
+
+	/* Get the 64 bit Unique Device Number UID */
+	/* The UID is used by firmware to derive */
+	/* 48-bit Device Address EUI-48 */
+	udn = LL_FLASH_GetUDN();
+
+	if (udn != 0xFFFFFFFF) {
+		/* Get the ST Company ID */
+		company_id = LL_FLASH_GetSTCompanyID();
+		/* Get the STM32 Device ID */
+		device_id = LL_FLASH_GetDeviceID();
+
+		/*
+		 * Public Address with the ST company ID
+		 * bit[47:24] : 24bits (OUI) equal to the company ID
+		 * bit[23:16] : Device ID.
+		 * bit[15:0] : The last 16bits from the UDN
+		 * Note: In order to use the Public Address in a final product, a dedicated
+		 * 24bits company ID (OUI) shall be bought.
+		 */
+
+		bd_addr_udn.val[0] = (uint8_t)(udn & 0x000000FF);
+		bd_addr_udn.val[1] = (uint8_t)((udn & 0x0000FF00) >> 8);
+		bd_addr_udn.val[2] = (uint8_t)device_id;
+		bd_addr_udn.val[3] = (uint8_t)(company_id & 0x000000FF);
+		bd_addr_udn.val[4] = (uint8_t)((company_id & 0x0000FF00) >> 8);
+		bd_addr_udn.val[5] = (uint8_t)((company_id & 0x00FF0000) >> 16);
+		bd_addr = &bd_addr_udn;
+	} else {
+		bd_addr = &bd_addr_dflt;
+	}
+
+	return bd_addr;
+}
+
+static int bt_hci_stm32wba_setup(const struct device *dev,
+				 const struct bt_hci_setup_params *params)
```

Comment:
Are you sure you want to ignore `params`? It may include a public address in case the application manually provided one through `bt_id_create()` before calling `bt_enable()`. You might want to override the on-device stored address with the app provided one in such a case.

Reply:
Hi @jhedberg,
 
It's a good remark, I could add something like this:
```
	if (bt_addr_eq(&(params->public_addr), BT_ADDR_ANY))  {
		param->value[0] = uid_addr->val[0];
		param->value[1] = uid_addr->val[1];
		param->value[2] = uid_addr->val[2];
		param->value[3] = uid_addr->val[3];
		param->value[4] = uid_addr->val[4];
		param->value[5] = uid_addr->val[5];
	}
	else {
		param->value[0] = params->public_addr.val[0];
		param->value[1] = params->public_addr.val[1];
		param->value[2] = params->public_addr.val[2];
		param->value[3] = params->public_addr.val[3];
		param->value[4] = params->public_addr.val[4];
		param->value[5] = params->public_addr.val[5];
	}
```
In this way the bt_id_create API can override value "assigned" to each board.
(It works at my side)

 What do you think ?

Reply:
> In this way the bt_id_create API can override value "assigned" to each board. (It works at my side)
> 
> What do you think ?

Looks fine to me, except for some minor style related things (e.g. I'd consider `memcpy()` or even `bt_addr_copy()` with a cast for param->value)

Reply:
Ok, we are aligned, I will rework my PR accordingly

Reply:
Hello @jhedberg
PR just updated 

Please review